### DISCUSSION TITLE
MSTUMPI

### DISCUSSION THREAD

**Discussion by th0ger** [TDAmeritrade/stumpy#1021] [Ideas]
Hey there, I'm new to STUMPY and just completed reading the educative and well-written tutorials.

I was wondering why there's no multi dimensional (MSTUMP) with incremental online support (STUMPI)?
I.e. an 'stumpy.mstumpi' function.

Seem like an obvious "gap" in the feature matrix.

**Comment by seanlaw** [comment]
@th0ger You are correct, there is no `MSTUMPI` equivalent. There was this [short discussion](https://github.com/TDAmeritrade/stumpy/issues/926) but there are a few reasons:

1. Multi-dimensional matrix profiles are hard to interpret "correctly" as it is different from stacking multiple one-dimensional matrix profiles on top of each other
2. Multi-dimensional matrix profiles are expensive/slow to compute 
3. We don't get many people using exploring multi-dimensional data (one-dimensional is already hard enough) that the majority of our energy is focused on improving the one-dimensional case
4. We try to faithfully reproduce the academic published work and so MSTUMPI was not something that was published (at least, not that we're aware of) so it was never addressed.

I think it is possible to do `MSTUMPI` but, honestly, I would rather spend time figuring out how to make `MSTUMP` faster first. At the end of the day, we are only a group of volunteers working on this on our own time but we welcome PR contributions or we can turn this into a feature enhancement request if you are truly interested in this feature?

**Comment by th0ger** [comment]
@seanlaw Thanks, it makes perfect sense. I did search for this before asking, but somehow missed the closed discussions.

From a learning perspective, it's good to know that mstumpi could be implemented.
I will not push you into prioritising this now.

I'll consider if it'll be possible for me to use STUMPI on each time series individually, and then combine the MPs in some sort of postprocessing.
